FAQ.md

```diff
@@ -115,12 +115,18 @@ $ rg --generate complete-fish > "$dir/rg.fish"
 
 For **zsh**:
 
-```
+```zsh
 $ dir="$HOME/.zsh-complete"
 $ mkdir -p "$dir"
 $ rg --generate complete-zsh > "$dir/_rg"
 ```
 
+or add the following to your `.zshrc`:
+
+```zsh
+$ source <(rg --generate complete-zsh) # note: this is about 4ms slower than the previous method
+```
```

Comment:
Why are we suggesting this method? I'm okay with supporting it as long as it doesn't have any costs to doing so, but I don't see a good reason to specifically highlight it in the docs.

Reply:
Personal preference.  I personally like doing it this way because it makes installation easier.  Since the generation is fast, I don't need to think about installing `_rg` and can just put in my zshrc

```zsh
  if type rg > /dev/null; then
	  eval "$(rg --generate=complete-zsh)"
  fi
```

I assume other people prefer it this way too, which is why projects like [fzf](https://github.com/junegunn/fzf?tab=readme-ov-file#setting-up-shell-integration) suggest it by default.  That is why I suggested it in the doc.

If you disagree, then I can remove this bit 🙂

Reply:
Based on how difficult it is to explain (see other thread), I think it makes more sense to show new users the easiest method by default, and if they want the more "idiomatic" / performant (slightly) option we can show it as an alternative.  The FAQ is going to be read by people who may not want to create directories, new files, and update their fpath.

Reply:
@okdana @BurntSushi how about I remove the FAQ patch, and we can discuss how to improve the FAQ in a new issue? (continued from https://github.com/BurntSushi/ripgrep/pull/2957#discussion_r1899864739 )